build.py

```diff
@@ -873,6 +872,13 @@ def create_dockerfile_buildbase(ddir, dockerfile_name, argmap):
 RUN pip3 install --upgrade pip && \
     pip3 install --upgrade wheel setuptools docker
 
+# Install boost version >= 1.78 for boost::span
+# Current libboost-dev apt packages are < 1.78, so install from tar.gz
+RUN wget -O /tmp/boost.tar.gz \
+        https://boostorg.jfrog.io/artifactory/main/release/1.80.0/source/boost_1_80_0.tar.gz && \
+    (cd /tmp && tar xzf boost.tar.gz) && \
+    mv /tmp/boost_1_80_0/boost /usr/include/boost
```

Comment:
You can also install boost library as a part of core repo CMake: https://github.com/triton-inference-server/python_backend/blob/54c693e5eac685ee343a3d11d4d37494e92286a3/CMakeLists.txt#L96-L105

Reply:
I tried copying what you did in Python backend, but ran into some issues with building. Build doesn't seem to find the boost install with `add_depencies(triton-core boostorg)` nor `find_package` after pulling boost 1.80 in through `ExternalProject_Add`. 

I'll leave this as-is for now in server repo to get working builds and focus on the functional tasks. I'll circle back to moving this to core repo build later.